frontend/src/plugins/impl/chat/chat-ui.tsx

```diff
@@ -441,6 +555,28 @@ export const Chatbot: React.FC<Props> = (props) => {
             .map((p) => p.text)
             .join("\n");
 
+          // Separate tool parts, reasoning parts, and other parts for assistant messages
+          const toolParts =
```

Comment:
we can share this with the chat sidebar `<RenderParts`. we may not have this now, but we can do some refactoring so that there is a generic component to render these parts and we don't need to unpack or adapt them